components/post_view/channel_intro_message/messages/standard.tsx

```diff
@@ -0,0 +1,211 @@
+                            name: (uiName),
+                            creator: (creatorName),
+                            date,
+                        }}
+                    />
+                </span>
+            );
+        }
+
+        if (channel.type === Constants.OPEN_CHANNEL) {
+            return (
+                <span>
+                    <FormattedMessage
+                        id='intro_messages.creator'
+                        defaultMessage='This is the start of the {name} channel, created by {creator} on {date}.'
+                        values={{
+                            name: (uiName),
+                            creator: (creatorName),
+                            date,
+                        }}
+                    />
+                </span>
+            );
+        }
+        return null;
+    }, [channel.create_at, channel.type, creatorName, locale, uiName]);
+
+    const purposeMessage = useMemo(() => {
+        if (!channel.purpose) {
+            return null;
+        }
+        if (channel.type === Constants.PRIVATE_CHANNEL) {
+            return (
+                <span>
+                    <FormattedMessage
+                        id='intro_messages.purposePrivate'
+                        defaultMessage=" This private channel's purpose is: {purpose}"
+                        values={{purpose: channel.purpose}}
+                    />
+                </span>
+            );
+        } else if (channel.type === Constants.OPEN_CHANNEL) {
+            return (
+                <span>
+                    <FormattedMessage
+                        id='intro_messages.purpose'
+                        defaultMessage=" This channel's purpose is: {purpose}"
+                        values={{purpose: channel.purpose}}
+                    />
+                </span>
+            );
+        }
+        return null;
+    }, [channel.purpose, channel.type]);
+
+    const isPrivate = channel.type === Constants.PRIVATE_CHANNEL;
+    let setHeaderButton = null;
+    const children = (
+        <SetHeaderButton
+            channel={channel}
+        />
+    );
+    if (children) {
+        setHeaderButton = (
+            <ChannelPermissionGate
+                teamId={channel.team_id}
+                channelId={channel.id}
+                permissions={[isPrivate ? Permissions.MANAGE_PRIVATE_CHANNEL_PROPERTIES : Permissions.MANAGE_PUBLIC_CHANNEL_PROPERTIES]}
+            >
+                {children}
+            </ChannelPermissionGate>
+        );
+    }
+
+    const boardCreateButton = (
+        <BoardsButton
+            channel={channel}
+            boardComponent={boardComponent}
+        />
+    );
+
+    const channelInviteButton = (
+        <AddMembersButton
+            totalUsers={totalUsers}
+            usersLimit={usersLimit}
+            channel={channel}
+            setHeader={setHeaderButton}
+            createBoard={boardCreateButton}
+        />
+    );
```

Comment:
apply the comments from other components here as well ... check for `channel.delete_at` beforehand and only render when needed

Reply:
done.  each xxIntroMessage component has a `renderButtons` boolean to check for if the buttons should render.